store/src/java/com/zimbra/cs/account/soap/SoapProvisioning.java

```diff
@@ -3224,5 +3239,43 @@ public List<HABGroupMember> getHABGroupMembers(Group group) throws ServiceExcept
     public void resetPassword(Account acct, String newPassword, boolean dryRun) throws ServiceException {
         throw ServiceException.UNSUPPORTED();
     }
+    
+    /**
+    *
+    * @param status reserved for future use if mail to send for different status to be added
+    * @param to accepts admins email address to which email has to be send
+    * @throws ServiceException if an error occurs while sending email for MDM notification mail
+    */
+    @Override
+    public String sendMdmEmail(String status, String to, String timeInterval) throws ServiceException {
+        String devicesInfo = MailConstants.HTML_DEVICE_TABLE_INFO_HEADER;
+        String emailBody = "";
+        String subject = LC.zimbra_mobile_mdm_notification_email_subject.value();
+        GetDeviceStatusResponse resp = invokeJaxb(new GetDeviceStatusRequest(null));
+        List<DeviceStatusInfo> statusOfDevices = resp.getDevices().stream().filter(devices -> DateTimeUtil.checkWithinTime(Timestamp.valueOf(devices.getTimestamp()), Integer.parseInt(timeInterval), TimeUnit.MINUTES))
+                .collect(Collectors.toList());
+        int j = 0;
+        for (int i = 0; i < statusOfDevices.size(); i++) {
+            if ((statusOfDevices.get(i).getStatus() == SyncAdminConstants.MDM_STATUS_SUSPENDED)) {
+                devicesInfo +=  MailConstants.HTML_TABLE_ROW +
+                        MailConstants.HTML_TABLE_DATA + (j + 1) + MailConstants.HTML_TABLE_DATA_CLOSE + 
+                        MailConstants.HTML_TABLE_DATA + statusOfDevices.get(i).getFriendlyName()+ MailConstants.HTML_TABLE_DATA_CLOSE +
+                        MailConstants.HTML_TABLE_DATA + statusOfDevices.get(i).getId() + MailConstants.HTML_TABLE_DATA_CLOSE +
+                        MailConstants.HTML_TABLE_DATA + statusOfDevices.get(i).getUserAgent() + MailConstants.HTML_TABLE_DATA_CLOSE +
+                        MailConstants.HTML_TABLE_DATA + statusOfDevices.get(i).getTimestamp() + MailConstants.HTML_TABLE_DATA_CLOSE +
+                        MailConstants.HTML_TABLE_ROW_CLOSE;
+                j++;
+            }
+        }
+        if (j == 0) {
+            emailBody = MailConstants.NO_QUARANTINED_LIST;
+            subject = MailConstants.NO_QUARANTINED_LIST;
+        } else {
+            emailBody = LC.zimbra_mobile_mdm_notification_email_body.value() + MailConstants.BREAK_LINE + MailConstants.BREAK_LINE
+                    + devicesInfo + MailConstants.HTML_TABLE_CLOSE;
+        }
+        SendEmailResponse sendEmailResponse = invokeJaxb(new SendEmailRequest(to, subject, emailBody));
```

Comment:
do we really need this new API `SendEmailRequest`? 
we could handle the logic here for sending email instead of calling the new api

Reply:
removed to from api only message to be taken from api